backend/src/controllers/auth-controller.ts

```diff
@@ -62,4 +66,80 @@ export const loginUser = async (req: Request, res: Response) => {
     } catch (error: any) {
         res.status(500).json({ message: 'Server error', error: error.message });
     }
-}
\ No newline at end of file
+}
+
+export const postReset = async (req: Request, res: Response) => {
+    try {
+        crypto.randomBytes(32, (err, buf) => {
+           if (err){
+               console.log(err);
+           }
+           const token = buf.toString('hex');
+           // @ts-ignore
+            User.findOne({email: req.body.email}).then(user => {
+               if(!user){
+                   console.log(err + ": No account with that email found");
+                   return res.redirect('/api/auth/');
+               }
+               user.resetToken = token;
+               // @ts-ignore
+               user.resetTokenExpiration = Date.now() + 3600000;
+               return user.save();
+           }).then(() => {
+               res.redirect('api/auth/login')
+               transporter.sendMail({
+                  to: req.body.email,
+                  from: 'arslankamcybekov7@gmail.com',
+                   subject: 'Password Reset Details',
+                   html: `
+                     <p>You requested a password reset</p>
+                     <p>Click this link <a> href="http://localhost:3000/reset/${token}">link</a> to set a new password</p>
+                   `
+               });
+           }).catch(err => {
+               console.log(err);
+           });
+        });
+        res.status(200).json({ message: 'Forgot password request received. Check your email for further instructions.' });
+    } catch (error: any) {
+        res.status(500).json({ message: 'Server error', error: error.message });
+    }
+};
+export const getPassword = async (req: Request, res: Response) => {
+    const token  = req.params.token;
+    User.findOne({resetToken: token, resetTokenExpiration: {$gt: Date.now()}}).then(user =>{
+        res.render('auth/new-password',
```

Comment:
Since this is an API for a frontend, we're not rendering any HTML using the backend.

You can remove this function and its corresponding route